stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/avatar/AvatarBitmapFactory.kt

```diff
@@ -0,0 +1,283 @@
+     *
+     * The [createUserBitmapBlocking] method takes precedence over this one if both are implemented.
+     * Prefer implementing this method if possible.
+     *
+     * @return The loaded bitmap or null if the loading failed (e.g. network issues).
+     */
+    public open suspend fun createUserBitmap(
+        user: User,
+        style: AvatarStyle,
+        @Px avatarSize: Int,
+    ): Bitmap? {
+        return StreamImageLoader.instance().loadAsBitmap(context, user.image)
+    }
+
+    /**
+     * Load a default Bitmap with the specified [avatarSize] to represent the [user].
+     * This should be a process that can never fail (e.g. not depend on network).
+     *
+     * This method takes precedence over [createDefaultUserBitmap] if both are implemented.
+     *
+     * Override this method only if you can't provide a suspending implementation, otherwise
+     * override [createDefaultUserBitmap] instead.
+     *
+     * @return The loaded bitmap.
+     */
+    public open fun createDefaultUserBitmapBlocking(
+        user: User,
+        style: AvatarStyle,
+        @Px avatarSize: Int,
+    ): Bitmap {
+        return NOT_IMPLEMENTED_MARKER
+    }
+
+    /**
+     * Load a Bitmap with the specified [avatarSize] to represent the [user], in a suspending operation.
+     *
+     * The [createDefaultUserBitmapBlocking] method takes precedence over this one if both are implemented.
+     * Prefer implementing this method if possible.
+     *
+     * @return The loaded bitmap.
+     */
+    public open suspend fun createDefaultUserBitmap(
+        user: User,
+        style: AvatarStyle,
+        @Px avatarSize: Int,
+    ): Bitmap {
+        return createInitialsBitmap(style, avatarSize, user.initials)
+    }
+
+    internal suspend fun createChannelBitmapInternal(
+
+        channel: Channel,
+        lastActiveUsers: List<User>,
+        style: AvatarStyle,
+        @Px avatarSize: Int,
+    ): Bitmap {
+        val customBitmap = withContext(DispatcherProvider.IO) {
+            createChannelBitmapBlocking(channel, lastActiveUsers, style, avatarSize)
+        }
+        val bitmap = when (customBitmap) {
+            NOT_IMPLEMENTED_MARKER -> createChannelBitmap(channel, lastActiveUsers, style, avatarSize)
+            else -> customBitmap
+        }
+
+        // Image loading successful, return the loaded bitmap
+        if (bitmap != null) {
+            return bitmap
+        }
+
+        // Use default, locally generated image instead
+        val customDefaultBitmap = withContext(DispatcherProvider.IO) {
+            createDefaultChannelBitmapBlocking(channel, lastActiveUsers, style, avatarSize)
+        }
+        return when (customDefaultBitmap) {
+            NOT_IMPLEMENTED_MARKER -> createDefaultChannelBitmap(channel, lastActiveUsers, style, avatarSize)
+            else -> customDefaultBitmap
+        }
+    }
+
+    public open fun createChannelBitmapBlocking(
```

Comment:
I'd like to have docs for all new public methods here, they're not completely trivial